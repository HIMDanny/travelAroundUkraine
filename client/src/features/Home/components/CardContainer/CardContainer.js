import React from 'react';
import { Link } from 'react-router-dom';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Navigation } from 'swiper';
import { Typography, styled, Container } from '@mui/material';
import { CardItem } from '..';
import { SliderButton } from '../../../../components';

import 'swiper/swiper.min.css';

const cards = [
  {
    name: 'Kyiv city walk',
    region: 'Kyiv region',
    id: 1,
    url: '',
  },
  {
    name: 'Lviv city walk',
    region: 'Lviv region',
    id: 2,
  },
  {
    name: 'Donetsk city walk',
    region: 'Donetsk region',
    id: 3,
  },
  {
    name: 'Odessa city walk',
    region: 'Odessa region',
    id: 4,
  },
  {
    name: 'Dnepr city walk',
    region: 'Dnepr region',
    id: 5,
  },
  {
    name: 'Chernivtsi city walk',
    region: 'Chernivtsi region',
    id: 6,
  },
  {
    name: 'Ternopil city walk',
    region: 'Ternopil region',
    id: 7,
  },
  {
    name: 'Lutsk city walk',
    region: 'Lutsk region',
    id: 8,
  },
];

const CardContainerHeaderContainer = styled('div')({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  marginTop: '30px',
});

const CardContainerSwiperContainer = styled('div')({
  position: 'relative',
  margin: '10px auto 0 auto',

  '@media (min-width: 300px)': {
    width: '265px',
  },

  '@media (min-width: 768px)': {
    width: '560px',
  },

  '@media (min-width: 1024px)': {
    width: '855px',
  },

  '@media (min-width: 1360px)': {
    width: '1150px',
  },
});

const CardContainer = () => (
  <Container>
    <CardContainerHeaderContainer>
      <Typography variant="h2" sx={{ marginBottom: '0px' }}>
        Popular tours
      </Typography>
      <Link to="/catalogue">
        <Typography>View All</Typography>
      </Link>
    </CardContainerHeaderContainer>
    <CardContainerSwiperContainer>
      <SliderButton id="previous" position={{ top: '44%', left: '-30px', rotate: '180deg' }} />
      <SliderButton id="next" position={{ top: '44%', right: '-30px' }} />
      <Swiper
        breakpoints={{
          265: {
            width: 265,
            slidesPerView: 1,
          },
          768: {
            width: 560,
            slidesPerView: 2,
          },
          1024: {
            width: 855,
            slidesPerView: 3,
          },
          1360: {
            width: 1150,
            slidesPerView: 4,
          },
        }}
        spaceBetween={30}
        slidesPerGroup={1}
        loop
        loopFillGroupWithBlank
        pagination={{
          clickable: true,
        }}
        navigation={{
          nextEl: '#next',
          prevEl: '#previous',
        }}
        modules={[Navigation]}
        className="mySwiper"
      >
        {cards.map(({ name, region, id }) => (
          <SwiperSlide key={id}>
            <CardItem name={name} region={region} id={id} />
          </SwiperSlide>
        ))}
      </Swiper>
    </CardContainerSwiperContainer>
  </Container>
);

export default CardContainer;
