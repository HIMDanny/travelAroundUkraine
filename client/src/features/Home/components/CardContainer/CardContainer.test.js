/* eslint-disable react/prop-types */
import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import CardContainer from './CardContainer';

const RouterWrapper = ({children}) => <MemoryRouter>{children}</MemoryRouter>;

jest.mock('@mui/material/Typography', () => ({children}) => <h2>{children}</h2>);
jest.mock('@mui/material/Container', () => ({children}) => <div>{children}</div>);

describe('Card Container Snapshot test', () => {
    test('should CardContainer match snapshot', () => {
        const {asFragment} = render(<CardContainer/>, {wrapper: RouterWrapper});
        expect(asFragment()).toMatchSnapshot();
    });
});
