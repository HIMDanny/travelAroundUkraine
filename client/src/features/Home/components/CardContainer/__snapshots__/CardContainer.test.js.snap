// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Container Snapshot test should CardContainer match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-4b3544"
    >
      <h2>
        Popular tours
      </h2>
      <a
        href="/catalogue"
      >
        <h2>
          View All
        </h2>
      </a>
    </div>
    <div
      class="css-gn8ri4"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-5jjyxr-MuiButtonBase-root-MuiIconButton-root"
        id="previous"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-fkmnqh-MuiButtonBase-root-MuiIconButton-root"
        id="next"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </button>
      <div
        class="swiper swiper-initialized swiper-horizontal swiper-pointer-events mySwiper"
      >
        <div
          class="swiper-wrapper"
          style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        />
        <div
          class="swiper-pagination"
        />
        0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
